Dockerfile.debian12

```diff
@@ -87,9 +87,10 @@ RUN apt-get update && apt-get install --no-install-recommends -y \
   python3-flake8 \
   python3-coverage \
   && \
-# colcon-devtools>0.2.3 is not released for debian12, so override it from pip
+  # colcon-devtools>0.2.3 is not released for debian12, so override it with pip
+  # typing_extensions needs to be >4.7.0, but is available on apt with 4.4.0 only
```

Comment:
So with this the CI build should be good, but the local build might have the same issue right?

Correct me if I'm wrong :)

Reply:
Sure!